datashader/layout.py

```diff
@@ -125,3 +125,67 @@ def __call__(self, nodes, edges, **params):
 
         # Return the nodes with updated positions
         return _merge_points_with_nodes(nodes, points)
+
+
+@nb.jit(nogil=True)
+def _create_random_xy_values(df, seed):
+    np.random.seed(seed)
+
+    points = np.asarray(np.random.random((len(df), 2)))
+    df['x'] = points[:, 0]
+    df['y'] = points[:, 1]
+    return df
```

Comment:
Why is this function using Numba?  It doesn't seem like there is anything for numba to do here.  If that's true, I'd move this code into ``random_layout.__call__`` directly.